### DISCUSSION TITLE
'Unknown column: Expr1' in previously working queries

### DISCUSSION THREAD

**Discussion by treesloth** [MarkMpn/Sql4Cds#213] [Q&A] [answered]
I recently updated SQL4CDS to the latest version.   Some queries that previously worked are now giving the error `Unknown column: Expr1`.  Here's an example:

UPDATE incident
SET    is_rapid_response = 1
WHERE  client_ticket = 'ABC123'
       AND is_rapid_response <> 1;

That's a trivial example, but it shows the idea.  The Execution Plan says that the error occurred in the Filter stage, but I don't see a way to get more information.  Removing the `AND is_rapid_response <> 1;` lets it work.  In fact, the `AND` portion of the `WHERE` clause now seems to cause this error generally.  Am I missing something, or has the update broken this?  Thank you.

**Comment by MarkMpn** [comment, marked as answer]
Thanks for reporting this. I've moved this to issue #214. From my investigation so far this seems to affect instances where you reference a column in both the SET clause and the WHERE clause, and you're not using a simple = operator on that column in the WHERE clause. So, assuming `is_rapid_response` is a boolean column, you could change your query to:

```sql
UPDATE incident
SET is_rapid_response = 1
WHERE client_ticket = 'ABC123'
AND (is_rapid_response = 0 OR is_rapid_response IS NULL);
```
